thrust/thrust/system/detail/generic/reduce.inl

```diff
@@ -58,15 +58,48 @@ reduce(thrust::execution_policy<ExecutionPolicy>& exec, InputIterator first, Inp
   return thrust::reduce(exec, first, last, init, ::cuda::std::plus<T>());
 } // end reduce()
 
-template <typename ExecutionPolicy, typename RandomAccessIterator, typename OutputType, typename BinaryFunction>
-_CCCL_HOST_DEVICE OutputType reduce(
-  thrust::execution_policy<ExecutionPolicy>&, RandomAccessIterator, RandomAccessIterator, OutputType, BinaryFunction)
+template <typename ExecutionPolicy, typename InputIterator, typename T, typename BinaryFunction>
+_CCCL_HOST_DEVICE T reduce(thrust::execution_policy<ExecutionPolicy>&, InputIterator, InputIterator, T, BinaryFunction)
 {
-  static_assert(thrust::detail::depend_on_instantiation<RandomAccessIterator, false>::value,
-                "unimplemented for this system");
-  return OutputType();
+  static_assert(thrust::detail::depend_on_instantiation<InputIterator, false>::value, "unimplemented for this system");
+  return T();
 } // end reduce()
 
+template <typename ExecutionPolicy, typename InputIterator, typename OutputIterator>
+_CCCL_HOST_DEVICE void reduce_into(
+  thrust::execution_policy<ExecutionPolicy>& exec, InputIterator first, InputIterator last, OutputIterator output)
+{
+  using T = thrust::detail::it_value_t<InputIterator>;
+
+  // use T(0) as init by default
+  thrust::reduce_into(exec, first, last, output, T(0));
+} // end reduce_into()
+
+template <typename ExecutionPolicy, typename InputIterator, typename OutputIterator, typename T>
+_CCCL_HOST_DEVICE void reduce_into(
+  thrust::execution_policy<ExecutionPolicy>& exec,
+  InputIterator first,
+  InputIterator last,
+  OutputIterator output,
+  T init)
+{
+  // use plus<T> by default
+  thrust::reduce_into(exec, first, last, output, init, ::cuda::std::plus<T>());
+} // end reduce_into()
+
+template <typename ExecutionPolicy, typename InputIterator, typename OutputIterator, typename T, typename BinaryFunction>
+_CCCL_HOST_DEVICE void reduce_into(
+  thrust::execution_policy<ExecutionPolicy>& exec,
+  InputIterator first,
+  InputIterator last,
+  OutputIterator output,
+  T init,
+  BinaryFunction binary_op)
+{
+  // use reduce by default
```

Comment:
Suggestion: Add a comment noting that `reduce_into` isn't specialized for the generic case, which is why this overload just calls into regular `reduce`.